### PR TITLE
KT-5193 Make multiline strings foldable

### PR DESCRIPTION
**Pull request by shiraji**
#KT-5193 Fixed

https://youtrack.jetbrains.com/issue/KT-5193

### CODE REVIEW

File: idea/src/org/jetbrains/kotlin/idea/KotlinFoldingBuilder.kt [hunk 1]
```diff
@@ -99,10 +100,18 @@ class KotlinFoldingBuilder : CustomFoldingBuilder(), DumbAware {
     override fun getLanguagePlaceholderText(node: ASTNode, range: TextRange): String = when {
         node.elementType == KtTokens.BLOCK_COMMENT -> "/${getFirstLineOfComment(node)}.../"
         node.elementType == KDocTokens.KDOC -> "/**${getFirstLineOfComment(node)}...*/"
+        node.elementType == KtNodeTypes.STRING_TEMPLATE -> "\"\"\"${getFirstLineOfString(node)}...\"\"\""
         node.psi is KtImportList -> "..."
         else ->  "{...}"
     }
 
+    private fun getFirstLineOfString(node: ASTNode): String {
+        val targetStringLine = node.text.split("\n").firstOrNull {
```
Comment (yole) [inline idea/src/org/jetbrains/kotlin/idea/KotlinFoldingBuilder.kt]: You can avoid multiple `replace` calls if you use `asSequence().map { it.replace(...) }.firstOrNull { it.isNotBlank }`

File: idea/src/org/jetbrains/kotlin/idea/KotlinFoldingBuilder.kt [hunk 2]
```diff
@@ -99,10 +100,18 @@ class KotlinFoldingBuilder : CustomFoldingBuilder(), DumbAware {
...
+            it.replace("\"\"\"", "").isNotBlank()
+        } ?: return ""
+        return " ${targetStringLine.replace("\"\"\"", "").trim()} "
```
Comment (yole) [inline idea/src/org/jetbrains/kotlin/idea/KotlinFoldingBuilder.kt]: It's better to add surrounding spaces in `getLanguagePlaceholderText()`, not in this function. (This applies also to `getFirstLineOfComment()`.)

Comment (shiraji) [inline idea/src/org/jetbrains/kotlin/idea/KotlinFoldingBuilder.kt]: But then if the string or the comment is empty, user will get `"""  ..."""`

File: idea/src/org/jetbrains/kotlin/idea/KotlinFoldingBuilder.kt (hunk 1 above)
Comment (shiraji) [inline idea/src/org/jetbrains/kotlin/idea/KotlinFoldingBuilder.kt]: Make sense. I will fix this.

File: idea/src/org/jetbrains/kotlin/idea/KotlinFoldingBuilder.kt (hunk 2 above)
Comment (yole) [inline idea/src/org/jetbrains/kotlin/idea/KotlinFoldingBuilder.kt]: Yes, you're right that the current solution produces somewhat cleaner results.

**Comment by yole** [comment]
Thanks for your contribution!

**Comment by shiraji** [comment]
Thanks for reviewing this pull request

**Pull request merged**